crates/client/rpc/src/versions/v0_8_0/methods/ws/lib.rs

```diff
@@ -0,0 +1,370 @@
+            starknet_core::types::BlockId::Number(block_n) => {
+                let err = || format!("Failed to retrieve block info for block {block_n}");
+                let block_latest = self
+                    .backend
+                    .get_block_n(&mp_block::BlockId::Tag(mp_block::BlockTag::Latest))
+                    .or_else_internal_server_error(err)?
+                    .ok_or(StarknetWsApiError::NoBlocks)?;
+
+                if block_n < block_latest.saturating_sub(BLOCK_PAST_LIMIT) {
+                    return Err(StarknetWsApiError::TooManyBlocksBack.into());
+                }
+
+                block_n
+            }
+            starknet_core::types::BlockId::Hash(block_hash) => {
+                let err = || format!("Failed to retrieve block info at hash {block_hash:#x}");
+                let block_latest = self
+                    .backend
+                    .get_block_n(&mp_block::BlockId::Tag(mp_block::BlockTag::Latest))
+                    .or_else_internal_server_error(err)?
+                    .ok_or(StarknetWsApiError::BlockNotFound)?;
+
+                let block_n = self
+                    .backend
+                    .get_block_n(&block_id)
+                    .or_else_internal_server_error(err)?
+                    .ok_or_else_internal_server_error(err)?;
+
+                if block_n < block_latest.saturating_sub(BLOCK_PAST_LIMIT) {
+                    return Err(StarknetWsApiError::TooManyBlocksBack.into());
+                }
+
+                block_n
+            }
+            starknet_core::types::BlockId::Tag(starknet_core::types::BlockTag::Latest) => self
+                .backend
+                .get_latest_block_n()
+                .or_internal_server_error("Failed to retrieve block info for latest block")?
+                .ok_or_internal_server_error("Failed to retrieve block info for latest block")?,
+            starknet_core::types::BlockId::Tag(starknet_core::types::BlockTag::Pending) => {
+                return Err(StarknetWsApiError::Pending.into());
+            }
+        };
+
+        let mut rx = self.backend.subscribe_block_info();
+        for n in block_n.. {
+            if sink.is_closed() {
+                return Ok(());
+            }
+
+            let block_info = match self.backend.get_block_info(&mp_block::BlockId::Number(n)) {
+                Ok(Some(block_info)) => {
+                    let err = format!("Failed to retrieve block info for block {n}");
+                    block_info.as_nonpending_owned().ok_or_internal_server_error(err)?
+                }
+                Ok(None) => break,
+                Err(e) => {
+                    let err = format!("Failed to retrieve block info for block {n}: {e}");
+                    return Err(StarknetWsApiError::internal_server_error(err).into());
+                }
+            };
+
+            send_block_header(&sink, block_info, block_n).await?;
+            block_n = block_n.saturating_add(1);
```

Comment:
i think you meant `n` here and not block_n?

Reply:
No, this should indeed be `block_n` so that we are left with the height of the latest block when exiting the loop